stubs/redis/redis/client.pyi

```diff
@@ -684,38 +684,297 @@ class PubSub:
     def run_in_thread(self, sleep_time=...): ...
     def ping(self, message: Optional[_Value] = ...) -> None: ...
 
-class BasePipeline:
+class Pipeline:
     UNWATCH_COMMANDS: Any
     connection_pool: Any
     connection: Any
     response_callbacks: Any
-    transaction: Any
+    transaction: bool
     shard_hint: Any
     watching: Any
     def __init__(self, connection_pool, response_callbacks, transaction, shard_hint) -> None: ...
-    def __enter__(self): ...
-    def __exit__(self, exc_type, exc_value, traceback): ...
-    def __del__(self): ...
-    def __len__(self): ...
+    def __enter__(self) -> Pipeline: ...
+    def __exit__(self, exc_type, exc_value, traceback) -> None: ...
+    def __del__(self) -> None: ...
+    def __len__(self) -> int: ...
     command_stack: Any
     scripts: Any
     explicit_transaction: Any
-    def reset(self): ...
+    def reset(self) -> None: ...
     def multi(self) -> None: ...
-    def execute_command(self, *args, **kwargs): ...
```

Comment:
`execute_command`, along with `parse_response`, `watch`, and `unwatch` were duplicated in the methods I copied from `Redis`, so I deleted these and updated the copied ones below.